File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/inbound/web/mission/dto/MissionResponse.java
package com.study.webflux.rag.infrastructure.inbound.web.mission.dto;

import com.study.webflux.rag.domain.mission.model.Mission;

public record MissionResponse(
	String missionId,
	String type,
	String name,
	String description,
	long rewardAmount,
	boolean repeatable
) {
	public static MissionResponse from(Mission mission) {
		return new MissionResponse(
			mission.missionId().value(),
			mission.type().name(),
			mission.name(),
			mission.description(),
			mission.rewardAmount(),
			mission.repeatable());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/inbound/web/mission/dto/MissionResponse.kt
package com.miyou.app.infrastructure.inbound.web.mission.dto

import com.miyou.app.domain.mission.model.Mission

data class MissionResponse(
    val missionId: String,
    val type: String,
    val name: String,
    val description: String?,
    val rewardAmount: Long,
    val repeatable: Boolean,
) {
    companion object {
        fun from(mission: Mission): MissionResponse =
            MissionResponse(
                mission.missionId.value,
                mission.type.name,
                mission.name,
                mission.description,
                mission.rewardAmount,
                mission.repeatable,
            )
    }
}
